numpy/core/tests/test_memmap.py

```diff
@@ -213,3 +215,9 @@ def test_empty_array(self):
 
         # ok now the file is not empty
         memmap(self.tmpfp, shape=(0,4), mode='w+')
+    
+    def test_shape_type(self):
+        memmap(self.tmpfp, shape=3, mode='w+')
+        memmap(self.tmpfp, shape=self.shape, mode='w+')
+        memmap(self.tmpfp, shape=list(self.shape), mode='w+')
+        memmap(self.tmpfp, shape=array(self.shape), mode='w+')
```

Comment:
```suggestion
        memmap(self.tmpfp, shape=asarray(self.shape), mode='w+')
```
(I would generally prefer to just use `np.` but the file uses it already.)